drivers/espi/espi_mchp_xec.c

```diff
@@ -0,0 +1,822 @@
+	case 20:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_20M;
+		break;
+	case 25:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_25M;
+		break;
+	case 33:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_33M;
+		break;
+	case 50:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_50M;
+		break;
+	case 66:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_66M;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	/* Set IO mode */
+	iomode = (iomode >> 1);
+	if (iomode > 3) {
+		return -EINVAL;
+	}
+
+	if (iomode != cur_iomode) {
+		cap1 &= ~MCHP_ESPI_GBL_CAP1_IO_MODE_MASK0 <<
+			MCHP_ESPI_GBL_CAP1_IO_MODE_POS;
+		cap1 |= (iomode << MCHP_ESPI_GBL_CAP1_IO_MODE_POS);
+	}
+	ESPI_CAP_REGS->GLB_CAP1 = cap1;
+
+	/* TODO: Consider to this later when configuring PC subdevices */
+	ESPI_HIO_BAR_REGS->IOBAR_INIT = MCHP_ESPI_IOBAR_INIT_DFLT;
+	ESPI_HIO_BAR_REGS->HOST_BAR_IOC |= MCHP_ESPI_IO_BAR_HOST_VALID;
+
+	/* Activate the eSPI block *.
+	 * Need to guarantee that this register is configured before RSMRST#
+	 * de-assertion and after pinmux
+	 */
+	ESPI_EIO_BAR_REGS->IO_ACTV = 1;
+	printk("eSPI block actived succesfully\n");
+	return 0;
+}
+
+static bool espi_xec_channel_ready(struct device *dev,
+	enum espi_channel ch)
+{
+	bool sts;
+
+	switch (ch) {
+	case ESPI_CHANNEL_PERIPHERAL:
+		sts = ESPI_CAP_REGS->PC_RDY & MCHP_ESPI_PC_READY;
+		break;
+	case ESPI_CHANNEL_VWIRE:
+		sts = ESPI_CAP_REGS->VW_RDY & MCHP_ESPI_VW_READY;
+		break;
+	case ESPI_CHANNEL_OOB:
+		sts = ESPI_CAP_REGS->OOB_RDY & MCHP_ESPI_OOB_READY;
+		break;
+	case ESPI_CHANNEL_FLASH:
+		sts = ESPI_CAP_REGS->FC_RDY & MCHP_ESPI_PC_READY;
+		break;
+	default:
+		sts = false;
+		break;
+	}
+	return sts;
+}
+
+static int espi_xec_read_request(struct device *dev,
+					struct espi_request_packet req)
+{
+	return -ENOTSUP;
+}
+
+static int espi_xec_write_request(struct device *dev,
+					struct espi_request_packet req)
+{
+	return -ENOTSUP;
+}
+
+static int espi_xec_send_vwire(struct device *dev,
+				enum espi_vwire_signal signal,
+				u8_t level)
+{
+	struct xec_signal signal_info = vw_tbl[signal];
+	u8_t xec_id = signal_info.xec_reg_idx;
+	u8_t src_id = signal_info.bit;
+
+	if ((src_id >= SRC_ID_MAX) ||
+		(xec_id >= ESPI_MSVW_IDX_MAX)) {
+		return -EINVAL;
+	}
+	if (signal_info.dir == M2S) {
+		ESPI_MSVW_REG *reg = &(ESPI_M2S_VW_REGS->MSVW00) + xec_id;
+
+		reg->SRC = (level << src_id);
+	}
+	if (signal_info.dir == S2M) {
+		ESPI_SMVW_REG *reg = &(ESPI_S2M_VW_REGS->SMVW00) + xec_id;
+
+		reg->SRC = (level << src_id);
+	}
+	return 0;
+}
+
+static int espi_xec_receive_vwire(struct device *dev,
+				enum espi_vwire_signal signal,
+				u8_t *level)
+{
+	struct xec_signal signal_info = vw_tbl[signal];
+	u8_t xec_id = signal_info.xec_reg_idx;
+	u8_t src_id = signal_info.bit;
+
+	if ((src_id >= SRC_ID_MAX) ||
+		(xec_id >= ESPI_SMVW_IDX_MAX) ||
+		(level == NULL)) {
+		return -EINVAL;
+	}
+	if (signal_info.dir == M2S) {
+		ESPI_MSVW_REG *reg = &(ESPI_M2S_VW_REGS->MSVW00) + xec_id;
+		*level = ((reg->SRC >> (src_id << 3)) & 0x01ul);
```

Comment:
it's better to have a self documented macro name instead of some "magic numbers" like here.